config/configmapconverter_test.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config // import "go.opentelemetry.io/collector/config"
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewRetrieved_NilConvertFunc(t *testing.T) {
+	mc := NewMapConverter(nil)
+	cfgMap := NewMap()
+	assert.NoError(t, mc.Convert(context.Background(), cfgMap))
+	assert.Len(t, cfgMap.AllKeys(), 0)
```

Comment:
This should probably test with a `Map` that has some random values in it.  The implementation could replace the provided map with a new empty map and still pass this test.

Reply:
Comment should be on the "dependent" PR.

Reply:
What?

Reply:
My bad, that is true, there is another "Empty" test in the dependent PR (which I updated :)) based on your comment). Will update this as well.

Reply:
Done, sorry again for the confusion.